packages/clients/test/sui/wallet.integration.test.ts

```diff
@@ -147,7 +149,9 @@ describe("ShinamiWallet", () => {
       beneficiary,
     );
     console.log("txDigest", txDigest);
-    expect(
+    await sui.waitForTransaction({ digest: txDigest });
+
+    await expect(
```

Comment:
I had tried this earlier but it still failed. I didn't realize that the await had to be on the expect, and not the signer.getBeneficiary

Reply:
I realized there were two separate issues here:

- Lack of `await` on the `expect().resolves`. This was causing the tests to NOT wait for the final verification before they wrap up. However, this by itself doesn't actually fail the tests, just causing mis-attribution of errors when a test actually fails (i.e. when test A failed, it may show up as test B failed instead, because the final verification of test A completed and threw while test B was already running).
  - FYI, `expect` has both sync and async forms. `expect(await somePromise).toBe(someValue)` is the sync form, which is roughly equivalent to `await expect(somePromise).resolves.toBe(someValue)`. Pay attention to what is passed to `expect()`.
- The actual test failures were caused by the lack of `sui.waitForTransaction`, which is described in the linked ticket.